persist/fs/retriever.go

```diff
@@ -313,6 +315,13 @@ func (r *blockRetriever) fetchBatch(
 
 		req.onRetrieved(seg)
 	}
+
+	err = seekerMgr.Return(shard, blockStart, seeker)
+	if err != nil {
+		r.fsOpts.InstrumentOptions().Logger().Errorf(
```

Comment:
Maybe put the logger on the `blockRetriever` struct so you can just `r.logger.Errorf(..)`?  There's only one per namespace so don't need to worry about saving space.

Reply:
can do